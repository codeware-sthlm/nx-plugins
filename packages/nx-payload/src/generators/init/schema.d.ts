export interface Schema {
  unitTestRunner?: 'jest' | 'none';
  skipFormat?: boolean;
  js?: boolean;
}
