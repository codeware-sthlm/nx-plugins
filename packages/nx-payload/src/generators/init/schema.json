{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxPayloadInit",
  "title": "Init Nx Payload Plugin",
  "type": "object",
  "properties": {},
  "required": []
}
