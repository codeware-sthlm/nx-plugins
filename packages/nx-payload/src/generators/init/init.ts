import {
  type GeneratorCallback,
  type Tree,
  addDependenciesToPackageJson,
  convertNxGenerator,
  formatFiles,
  runTasksInSerial
} from '@nx/devkit';
import { initGenerator as expressInitGenerator } from '@nx/express/src/generators/init/init';

import { payloadPluginsVersions, payloadVersion } from '../../utils/versions';

import type { Schema } from './schema';

function updateDependencies(tree: Tree) {
  return addDependenciesToPackageJson(
    tree,
    { ...payloadPluginsVersions, payload: payloadVersion },
    {}
  );
}

/**
 * Add required application dependencies
 */
export async function initGenerator(
  tree: Tree,
  schema: Schema
): Promise<GeneratorCallback> {
  const tasks: GeneratorCallback[] = [];

  const expressTask = await expressInitGenerator(tree, schema);
  tasks.push(expressTask);

  const installTask = updateDependencies(tree);
  tasks.push(installTask);

  await formatFiles(tree);

  return runTasksInSerial(...tasks);
}

export default initGenerator;
export const initSchematic = convertNxGenerator(initGenerator);
