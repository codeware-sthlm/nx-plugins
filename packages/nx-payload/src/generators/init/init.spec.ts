import { type Tree, addDependenciesToPackageJson, readJson } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import type { PackageJson } from 'nx/src/utils/package-json';

import { initGenerator } from './init';

describe('init', () => {
  let tree: Tree;

  console.log = jest.fn();
  console.warn = jest.fn();

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
  });

  it('should generate correct package.json', async () => {
    await initGenerator(tree, {});
    const packageJson = readJson<PackageJson>(tree, 'package.json');
    expect(packageJson).toMatchSnapshot();
  });

  it('should keep existing dependencies', async () => {
    const existing = 'existing';
    const existingVersion = '1.0.0';
    const dependencies = {
      [existing]: existingVersion
    };
    const devDependencies = {
      [existing]: existingVersion
    };
    addDependenciesToPackageJson(tree, dependencies, devDependencies);

    await initGenerator(tree, {});
    const packageJson = readJson<PackageJson>(tree, 'package.json');

    expect(packageJson.dependencies[existing]).toBeDefined();
    expect(packageJson.devDependencies[existing]).toBeDefined();
  });
});
