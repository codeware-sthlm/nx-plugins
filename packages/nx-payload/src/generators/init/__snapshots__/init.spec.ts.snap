// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init should generate correct package.json 1`] = `
{
  "dependencies": {
    "@payloadcms/bundler-webpack": "^1.0.6",
    "@payloadcms/db-mongodb": "^1.4.1",
    "@payloadcms/db-postgres": "^0.5.2",
    "@payloadcms/richtext-slate": "^1.3.1",
    "express": "^4.18.1",
    "payload": "^2.8.2",
    "process": "latest",
    "tslib": "^2.4.1",
  },
  "devDependencies": {
    "@nx/express": "19.8.4",
    "@types/express": "latest",
    "rimraf": "latest",
  },
  "name": "@proj/source",
}
`;
